server.go

```diff
@@ -249,19 +274,82 @@ func NewFromConfig(conf Config) (ret Server, err error) {
 	conf.TLSKey = REDACTED
 	log.WithField("config", conf).Debug("Initialized server")
 
-	if len(conf.TraceAddress) > 0 && (conf.TraceAPIAddress != "") {
+	// Configure tracing workers and sinks
+	if len(conf.TraceAddress) > 0 && ret.tracingSinkEnabled() {
 
 		ret.TraceWorker = NewTraceWorker(ret.Statsd)
 
 		ret.TraceAddr, err = net.ResolveUDPAddr("udp", conf.TraceAddress)
-		log.WithField("traceaddr", ret.TraceAddr).Info("Set trace address")
+		log.WithField("traceaddr", ret.TraceAddr).Info("Listening for trace spans on address")
+
 		if err == nil && ret.TraceAddr == nil {
 			err = errors.New("resolved nil UDP address")
 		}
 		if err != nil {
 			return
 		}
 		trace.Enable()
+
+		// configure Datadog as sink
+		if ret.DDTraceAddress != "" {
+			ret.tracerSinks = append(ret.tracerSinks, tracerSink{
+				name:   "Datadog",
+				tracer: nil,
+				flush:  flushSpansDatadog,
+			})
+			log.Info("Configured Datadog trace sink")
+		}
+
+		// configure Lightstep as Sink
+		if ret.traceLightstepAccessToken != "" {
+			resolved, err := resolveEndpoint(conf.TraceLightstepCollectorHost)
+			if err != nil {
+				log.WithError(err).WithFields(logrus.Fields{
+					"host": conf.TraceLightstepCollectorHost,
+				}).Error("Error resolving Lightstep collector host")
+				return ret, err
+			}
+
+			host, err := url.Parse(resolved)
+			if err != nil {
+				log.WithError(err).WithFields(logrus.Fields{
+					"host":     conf.TraceLightstepCollectorHost,
+					"resolved": resolved,
+				}).Error("Error parsing Lightstep collector URL")
+				return ret, err
+			}
+
+			port, err := strconv.Atoi(host.Port())
+			if err != nil {
+				port = lightstepDefaultPort
+			}
+
+			log.WithFields(logrus.Fields{
+				"Host": host.Hostname(),
+				"Port": port,
+			}).Info("Dialing lightstep host")
```

Comment:
I was using it when debugging which host and port Veneur was using to dial the Lightstep collectors, and it actually came in handy again today with a hiccup we had in QA.